client/src/common/components/Dropdown.jsx

```diff
@@ -15,15 +16,27 @@ class Dropdown extends PureComponent {
     this.dropdown = React.createRef();
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps) {
     const { isVisible } = this.state;
+    const {
+      location: { pathname }
+    } = this.props;
+    const {
+      location: { pathname: prevPathname }
+    } = prevProps;
```

Comment:
I'm still not sure if we should bind the dropdown component with additional libraries. With this solution, we will be able to use it only in apps that use react-router and only there, where changing the route should close the dropdown.

Reply:
As I said, and Przemo agreed with my it is a bad practise to control visibility of component in two places (inside component and additional prop from outside).

I know what you mean, and I think if someone want to use this component in future, and he will copy and paste this it will be clear thath he has to remove with router statement and componentDidUpdate method.

And the one last thing, in my opinion this component now is designed in a wrong way.If we want to make generic components, that can be used across many other projects we should do this in a diffrent way. It should aways take visbility information from outside, from parrent component. It should have isVisible property which means that developer has to provide this property from outside.

```

<Button onClick={isVisible ? showDropdown : hideDropdown } />
<Dropdown isVisible={true || false}>
   {children}
</Dropdown>
```

and in this way, dropdown component is as Dan Abramov said "dump" functional component responsible only for rendering data, but he don't care anything about internal state etc.
I know also what was the idea of working component "out of the box", but I don't think it was done properly. 
For now I in my opinion it works as expected. And I totally agree with you @alemikolo , that it's not generic

Reply:
The second solution to work this out of the box is to create two dependent components.
DropdownController which will have internal state to control the visibility of the Dropdown component. And a Dropdown component will still be as I said before a "dump" functional component to render data only.

And in this way, we have "out of the box" dropdown solution, Where they can be used together or just Dropdown component independently. 

```
<DropdownController>
    <button onClick={isVisible ? showDropdown : hideDropdown}>

    <Dropdown isVisible={}>
          {children}
    <Dropdown>
<DropdownController>
```